spec/models/expense_spec.rb

```diff
@@ -9,10 +9,14 @@
     it { should have_db_column :other }
     it { should have_db_column :insurance }
   end
+  
+  describe 'Associations' do
+    it { should belong_to :user}
+  end
 
   describe 'Factory' do
     it 'should have valid Factory' do
-      expect(FactoryBot.create(:expense)).to be_valid
+      expect(FactoryBot.create(:user)).to be_valid
```

Comment:
Why do you test the factorybot validation of user in the income and expense model specs when you already test that in the user model spec?

Reply:
When we don't have the user validation there we get this error message:
```
1) Income Factory should have valid Factory
     Failure/Error: expect(FactoryBot.create(:income)).to be_valid
     
     ActiveRecord::RecordInvalid:
       Validation failed: User must exist

```

Reply:
```
user = FactoryBot.create(:user)
expect(FactoryBot.create(:expense, user: user)).to be_valid
``` 
Would be one way to do it.